apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/app/GenesisImporter.java

```diff
@@ -404,24 +425,44 @@ private int savePublicKeys(final TransactionalDataSource dataSource) {
             (System.currentTimeMillis() - start) / 1000, totalAmount
         );
 
-        validateBalanceNumber(count);
+        try {
+            validateBalanceNumber(count);
+        } catch (GenesisImportException e) {
+            throw new RuntimeException(e);
+        }
 
         return Pair.of(totalAmount, count);
     }
 
-    List<Map.Entry<String, Long>> loadGenesisAccounts() {
+    List<Map.Entry<String, Long>> loadGenesisAccounts() throws GenesisImportException {
         final String path = blockchainConfig.getChain().getGenesisLocation();
         log.debug("Genesis accounts json resource path = " + path);
+        final InputStream is = resourceLocator.locate(path).orElseThrow(() -> new RuntimeException("The resource could not be found, path=" + path));
+
+        final Queue<Map.Entry<String, Long>> sortedEntries = loadGenesisAccountsFromIS(is);
 
+        final int balanceNumber = sortedEntries.size();
+        validateBalanceNumber(balanceNumber);
+
+        return sortedEntries.stream()
+            .skip(1) //skip first account to collect only genesis accounts
```

Comment:
What does happen on 'skip(1)' if 'sortedEntries' is empty ? may be something like .orElseThrow(...) is needed for such case?

Reply:
There is always at least one entry.